docs/installation/local.md

```diff
@@ -2,169 +2,177 @@
+- Tip:
 
-# System Wide Installation
+  + Setup SSH key in your profile, and use SSH method to clone the source code, so that you don't have to type password repeatly.
 
-```sh
-sudo -H pip3 install -r requirements.txt
-```
-hint: You may need to upgrade your pip version and install following packages if you encounter errors while installing the requirements.
+  + To let your personal fork up-to-date with the original FOSSASIA repo, add original repo to "remote" list and regularly fetch its new content.
 
-**Note:** For Mac OS Sierra users, if you get an error that 'openssl/aes.h' could not be found when installing requirements.txt using pip, try the steps shown here - [OSX openssl header error](https://tutorials.technology/solved_errors/1-OSX-openssl_opensslv_h-file-not-found.html)
+  ```sh
+  git remote add upstream https://github.com/fossasia/open-event-server.git
+  git fetch upstream
+  git merge upstream/development
+  ```
 
-# Installation in Virtual Environment
+## Install system dependencies
 
-You can use either **pip** or **pipenv** to install Open Event Server in a virtual environment.
+These are softwares on which our Open Event server depends, and C-based libraries on which our Python packages depend.
 
-Firstly, open a terminal and enter
+### For Mac OS
 
 ```sh
-# For linux users
-sudo apt-get install python3-dev
-sudo apt-get install libpq-dev
-sudo apt-get install libffi6 libffi-dev
-
-# For macOS users
+brew install postgresql
 brew install python@3
 brew install libmagic
+brew install redis
 ```
 
-## Using pip and virtualenv
+**Note:** For Mac OS Sierra users, if you get an error that 'openssl/aes.h' could not be found when installing requirements.txt using pip, try the steps shown here - [OSX openssl header error](https://tutorials.technology/solved_errors/1-OSX-openssl_opensslv_h-file-not-found.html)
+
+### For Debian/Ubuntu
 
-Open a terminal and enter the following commands to setup a virtual environment
+Enter the project source folder and run:
 
 ```sh
-sudo apt-get install python3.7
-virtualenv -p python3.7 venv
-. venv/bin/activate
+xargs -a deb-packages.txt sudo apt install
 ```
 
-Now to install the dependencies using pip, type
+In case you use Ubuntu 20.04+, where Python 3.7 is not provided in official repo, you can use [pyenv](https://github.com/pyenv/pyenv) to install Python 3.7 (Open Event Server is not compatible with Python 3.8+ yet).
 
-```sh
-pip3 install -r requirements.txt
-```
+## Create a Python virtual environment
 
-## Using pipenv
+Python virtual environment is a way to keep this project's collection of libraries not interfere with other projects or the system. There are many ways to create Python virtual environment (all have the same core), like:
 
-Using pipenv, you will not need to set up virtualenv. It will do it automatically for you
+- Use Python built-in tool:
 
-To setup a virtual environment and install the dependices, enter in a terminal
+  ```sh
+  python3 -m venv my-project-name
```

Comment:
```suggestion
  python3 -m venv .venv
```

.venv is a recognized and gitignored folder

Reply:
I'm against placing virtual environment inside source folder, with explanation included in README.
In summary, if doing so, when you do some action involving scan current folder, you will accidentality scan files in the virtual environment as well, which slow down your task, or worse, can cause some unexpected side effect.

Reply:
I think it's more of personal preference. I personally prefer to have it inside the source directory so that if I ever want to see how things are implemented, I have easy access to the libraries it is using. Especially because we are using a fork of `flask-rest-jsonapi` which is maintained by FOSSASIA itself.

Reply:
But your command here will create the environment variable in the project as well. Here, the only change I am suggesting is to use a standard name. You have mentioned your preferred method below which if contributors want, can use. The above command will be directly be copied by students and lo and behold, they'll have a `my-project-name` folder in their repo which they'll commit in their next PR with all the libraries they installed in it. Besides, having separate folder of virtual environments:

1. Requires 3rd party tools not available in python distribution and learn its syntax
2. Difficult to manage as instead of running a project local command `source .venv/bin/activate`, they have to run different commands for different projects depending on the tool of choice
3. Makes it difficult for us to write guides or help users as it is not as simple as to just say to run `source .venv/bin/activate`

But anyway, you already have written your recommendation of virtualenv below and should the student choose it, they can use those instructions. But this instruction is using the previous method so at least it should be causing least friction by using a standard .gitignored name

Reply:
@prateekj117  @iamareebjamal 

> they'll have a my-project-name folder in their repo

Sorry, when I write down "python3 -m venv my-project-name", I think about that user should run it outside the repo. But everyone read here will think about running it inside the repo. I'm wrong here. I agree to change to `python3 -m venv .venv`.